#
# <meta:header>
#   <meta:licence>
#     Copyright (c) 2023, ROE (http://www.roe.ac.uk/)
#
#     This information is free software: you can redistribute it and/or modify
#     it under the terms of the GNU General Public License as published by
#     the Free Software Foundation, either version 3 of the License, or
#     (at your option) any later version.
#
#     This information is distributed in the hope that it will be useful,
#     but WITHOUT ANY WARRANTY; without even the implied warranty of
#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#     GNU General Public License for more details.
#
#     You should have received a copy of the GNU General Public License
#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
#   </meta:licence>
# </meta:header>
#
#zrq-notes-time
#zrq-notes-indent
#zrq-notes-crypto
#zrq-notes-ansible
#zrq-notes-osformat
#zrq-notes-zeppelin
#

    Target:

        New deployment to test the data in Echo S3.

    Result:

        Work in progress ...


# -----------------------------------------------------
# Check which cloud is currently live.
#[user@laptop]

    ssh fedora@live.gaia-dmp.uk \
        '
        date
        hostname
        '

    >   Tue 10 Jan 17:07:57 UTC 2023
    >   iris-gaia-blue-20221013-zeppelin


# -----------------------------------------------------
# Create a container to work with.
#[user@laptop]

    #
    # Live is blue, selecting green for the deployment.
    #

    source "${HOME:?}/aglais.env"

    agcolour=green
    configname=zeppelin-54.86-spark-6.26.43

    clientname=ansibler-${agcolour}
    cloudname=iris-gaia-${agcolour}

    podman run \
        --rm \
        --tty \
        --interactive \
        --publish  "3000:3000" \
        --name     "${clientname:?}" \
        --hostname "${clientname:?}" \
        --env "cloudname=${cloudname:?}" \
        --env "configname=${configname:?}" \
        --env "SSH_AUTH_SOCK=/mnt/ssh_auth_sock" \
        --volume "${SSH_AUTH_SOCK:?}:/mnt/ssh_auth_sock:rw,z" \
        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
        ghcr.io/wfau/atolmis/ansible-client:2022.07.25 \
        bash

    >   ....
    >   ....

# -----------------------------------------------------
# Deploy everything.
#[root@ansibler]

    time \
        source /deployments/hadoop-yarn/bin/deploy.sh

    >   ....
    >   ....
    >   Unable to establish connection to https://arcus.openstack.hpc.cam.ac.uk:5000/v3/auth/tokens: HTTPSConnectionPool(host='arcus.openstack.hpc.cam.ac.uk', port=5000): Max retries exceeded with url: /v3/auth/tokens (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f5dea5011b0>: Failed to establish a new connection: [Errno 111] Connection refused'))
    >   ....
    >   ....

    #
    # Failed with connection errors from ROE Eduroam to Cambridge cloud.
    #

    #
    # Try again running the container from our desktop in Devon.
    #

# -----------------------------------------------------
# Remote connection to desktop in Devon.
#[user@laptop]

    ssh methionine

# -----------------------------------------------------
# Create a container to work with.
#[user@desktop]

    #
    # Live is blue, selecting green for the deployment.
    #

    source "${HOME:?}/aglais.env"

    agcolour=green
    configname=zeppelin-54.86-spark-6.26.43

    clientname=ansibler-${agcolour}
    cloudname=iris-gaia-${agcolour}

    podman run \
        --rm \
        --tty \
        --interactive \
        --publish  "3000:3000" \
        --name     "${clientname:?}" \
        --hostname "${clientname:?}" \
        --env "cloudname=${cloudname:?}" \
        --env "configname=${configname:?}" \
        --env "SSH_AUTH_SOCK=/mnt/ssh_auth_sock" \
        --volume "${SSH_AUTH_SOCK:?}:/mnt/ssh_auth_sock:rw,z" \
        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
        ghcr.io/wfau/atolmis/ansible-client:2022.07.25 \
        bash

    >   ....
    >   ....


# -----------------------------------------------------
# Deploy everything.
#[root@ansibler]

    time \
        source /deployments/hadoop-yarn/bin/deploy.sh

    >   ....
    >   ....
    >   TASK [Create a symlink for the Hadoop version] *********************************
    >   fatal: [master01]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh_dispatch_run_fatal: Connection to UNKNOWN port 65535: Broken pipe", "unreachable": true}
    >   fatal: [worker03]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh_dispatch_run_fatal: Connection to UNKNOWN port 65535: Broken pipe", "unreachable": true}
    >   fatal: [worker04]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh_dispatch_run_fatal: Connection to UNKNOWN port 65535: Broken pipe", "unreachable": true}
    >   fatal: [worker02]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh_dispatch_run_fatal: Connection to UNKNOWN port 65535: Broken pipe", "unreachable": true}
    >   fatal: [worker01]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh_dispatch_run_fatal: Connection to UNKNOWN port 65535: Broken pipe", "unreachable": true}
    >   changed: [zeppelin]
    >   changed: [worker05]
    >   changed: [worker06]
    >   ....
    >   ....

    #
    # Failed due to network errors in the cloud.
    # Seen similar errors in notes/zrq/20220725-02-red-deploy.txt.
    #


